### PR TITLE
Refactored map-menu implementation for flexible theme development

### PR DESCRIPTION
**Pull request by winstonwumbo**
## New Features
* Implemented new `LitElement` properties to track orientation and depth states
* New `HAXCMSFlexMenu` function call added to `HAXCMSMobileMenu` coreutil, hooks with the `responsiveSize` utility
* `Map-menu` responsively adapts between a horizontal desktop state and vertical mobile state
* CSS `Grid` and `Flexbox` used to dynamically align menu items according to available columns
* CSS variables added for each layer to expand customization and fine tune formatting for unique themes
* Conditional CSS and web component rendering to maintain backwards compatibility with legacy vertical `map-menu`
## Related Issue(s)
* haxtheweb/issues#2023

### CODE REVIEW

File: elements/map-menu/map-menu.js [hunk 1]
```diff
@@ -138,13 +347,16 @@ class MapMenu extends LitElement {
    * LitElement life cycle - render
    */
   render() {
+    console.log("Here is flex: ", this.isFlex);
```
Comment (btopro) [inline elements/map-menu/map-menu.js:350]: console log

File: elements/map-menu/map-menu.js [hunk 2]
```diff
@@ -234,6 +459,22 @@ class MapMenu extends LitElement {
       }
     });
   }
+
+  __openChanged(e){
+    e.stopImmediatePropagation();
+    e.preventDefault();
+    e.stopPropagation();
+    if(this.isHorizontal){
```
Comment (btopro) [inline elements/map-menu/map-menu.js:467]: test for this.shadowRoot as well just to make me happy. Possible event is set in constructor, fires due to low performing page load and then does not have root yet. Highly unlikely but still

File: elements/map-menu/lib/map-menu-submenu.js [hunk 3]
```diff
@@ -118,11 +135,22 @@ class MapMenuSubmenu extends LitElement {
     }, 0);
   }
 
-  __active() {
+  __active(e) {
+    console.log(e.type);
```
Comment (btopro) [inline elements/map-menu/lib/map-menu-submenu.js:139]: console

**Review by btopro — CHANGES_REQUESTED** [review]

File: elements/map-menu/map-menu.js (hunk 1 above)
Comment (winstonwumbo) [reply elements/map-menu/map-menu.js]: Removed

File: elements/map-menu/map-menu.js (hunk 2 above)
Comment (winstonwumbo) [reply elements/map-menu/map-menu.js]: A check has been added to the if statement

File: elements/map-menu/lib/map-menu-submenu.js (hunk 3 above)
Comment (winstonwumbo) [reply elements/map-menu/lib/map-menu-submenu.js]: Removed

**Pull request merged**